src/wp-includes/rest-api/class-wp-rest-server.php

```diff
@@ -1291,10 +1291,22 @@ public function get_index( $request ) {
 			$fields[] = '_embedded';
 		}
 
-		if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
+		$links_or_embedded_included = rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields );
+
+		$site_logo_included     = rest_is_field_included( 'site_logo', $fields );
+		$site_icon_included     = rest_is_field_included( 'site_icon', $fields );
+		$site_icon_url_included = rest_is_field_included( 'site_icon_url', $fields );
+
+		if ( $links_or_embedded_included ) {
```

Comment:
I'm thinking it might be more performant (tiny micro-optimization) and more consistent with other instances to refactor this into if / else if / else if approach.

```
if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
	$response->add_link( 'help', 'https://developer.wordpress.org/rest-api/' );
	$this->add_active_theme_link_to_index( $response );
	$this->add_site_logo_to_index( $response );
	$this->add_site_icon_to_index( $response );
} else if ( rest_is_field_included( 'site_logo', $fields ) ) {
	$this->add_site_logo_to_index( $response );
} else if ( rest_is_field_included( 'site_icon', $fields ) || rest_is_field_included( 'site_icon_url', $fields ) ) {
	$this->add_site_icon_to_index( $response );
}
```

In doing so:

* Performance by avoiding assignments, memory for the variables, and rechecking the `$links_or_embedded_included` value for each of the `if` conditionals.
Granted, the trade-off is redundant code. But I think it's okay to repeat since the goal of the original changeset was for performance.

* The change is less, as the new change shifts to only adding the 2 `else if` branches while retaining the original `if` branch.

Reply:
Thank you for the review, @hellofromtonya. I appreciate it.
This code is definitely more performant.
However, it doesn't account for cases when only the `site_logo` and `site_icon` fields are requested because only the first `else if` expression that evaluates to true would be executed.
In that case, only the `site_logo` field will be included in the response.
I've updated the unit test to cover that use case as well.
I'm considering the following modification to address this issue:
```php
if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
	$response->add_link( 'help', 'https://developer.wordpress.org/rest-api/' );
	$this->add_active_theme_link_to_index( $response );
	$this->add_site_logo_to_index( $response );
	$this->add_site_icon_to_index( $response );
} else {
	if ( rest_is_field_included( 'site_logo', $fields ) ) {
		$this->add_site_logo_to_index( $response );
	}
	if ( rest_is_field_included( 'site_icon', $fields ) || rest_is_field_included( 'site_icon_url', $fields ) ) {
		$this->add_site_icon_to_index( $response );
	}
}
```

What do you think?

Reply:
Aha you're right. Yes, I think that's a good approach.